File: core/src/main/java/forpleuvoir/dhwu1a/core/message/StrangerMessage.java
package forpleuvoir.dhwu1a.core.message;

import com.google.gson.JsonObject;
import forpleuvoir.dhwu1a.core.message.base.Message;
import forpleuvoir.dhwu1a.core.message.base.MessageType;
import forpleuvoir.dhwu1a.core.message.messagesender.StrangerMessageSender;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message
 * <p>#class_name StrangerMessage
 * <p>#create_time 2021/7/1 22:18
 */
public class StrangerMessage extends Message {


    public StrangerMessage(JsonObject object) {
        super(MessageType.StrangerMessage, object);
    }

    @Override
    public void printMessageLog() {
        log.info("M/Bot.{} 陌生人[{}({})] >> {}", bot.id, ((StrangerMessageSender) sender).nickname, sender.id,
                 toPlainText()
        );
    }

}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/StrangerMessage.kt
package forpleuvoir.dhwu1a.core.message

import com.google.gson.JsonObject
import forpleuvoir.dhwu1a.core.message.base.Message
import forpleuvoir.dhwu1a.core.message.base.MessageType
import forpleuvoir.dhwu1a.core.message.messagesender.StrangerMessageSender

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message
 *
 * #class_name StrangerMessage
 *
 * #create_time 2021/7/1 22:18
 */
class StrangerMessage(`object`: JsonObject) : Message(MessageType.StrangerMessage, `object`) {
    override fun printMessageLog() {
        log.info(
            "M/Bot.{} 陌生人[{}({})] >> {}", bot.id, (sender as StrangerMessageSender).nickname, sender.id,
            toPlainText()
        )
    }
}